### ISSUE TITLE
Rule Request: unused value in capture group

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#2715]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

Similar to unused parameter in closure.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

```swift
// Triggering
{ [foo] in _ }()

// Non- Triggering
{ [foo] in foo.bar() }()
```

3. Should the rule be configurable, if so what parameters should be configurable?

Only severity.

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

Enabled by default, same as `unused_closure_parameter`.

**Comment by AngryBaby** [comment]
it work!
